AirCasting/CoreData/MeasurementStreamStorage.swift

```diff
@@ -24,8 +24,8 @@ protocol MeasurementStreamStorageContextUpdate {
 }
 
 extension HiddenCoreDataMeasurementStreamStorage {
-    func addMeasurementValue(_ value: Double, at location: CLLocationCoordinate2D? = nil, toStreamWithID id: MeasurementStreamLocalID) throws {
-        try addMeasurement(Measurement(time: Date(), value: value, location: location), toStreamWithID: id)
+    func addMeasurementValue(_ value: Double, at location: CLLocationCoordinate2D? = nil, toStreamWithID id: MeasurementStreamLocalID, on time: Date? = Date()) throws {
+        try addMeasurement(Measurement(time: time!, value: value, location: location), toStreamWithID: id)
```

Comment:
wouldn't it be better to not make the argument an optional in the first place?

Reply:
I agree. Is there a reason for making it optional?

Reply:
There is no, you are right